### PR TITLE
Create verification.md

### PR DESCRIPTION
**Pull request by kimdhamilton**
Attempt at a human-readable description of what I'm currently calling "Verification, Validation, and Veracity", informed by VC-DATA-MODEL definitions and other sources including vc-http-api, vc-js implementation, etc.

The idea is for this to provide conceptual buckets to help people understand the "3V" process, but also be reasonably (long-term) unchanging. Ideally it will also help provide a framework to distinguish checks performed by this repo vs checks that should be performed by callers.

### CODE REVIEW

**Review by mprorock — APPROVED** [review]
This is awesome! Thank you

File: docs/verification.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    - See "Verify(VC)"
+3. 3V VP outer layer
+    - See "Verify(VP)"
+
+## Verification
+
+### Verify (VC)
+
+1. Check well-formed according to VC-DATA-MODEL
+    1. E.g. are required fields present
```
Comment (TallTed) [inline docs/verification.md:59]:
```suggestion
    1. E.g., are required fields present?
```

File: docs/verification.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
+
+# Verification, Validation, and Veracity
+
+This attempts a general, user-facing description of the process whereby a relying party checks whether they will accept VC or VP. An additional goal is to provide a framework and context for the “verification” steps provided in the vc-http-api repository.
```
Comment (msporny) [inline docs/verification.md:4]:
```suggestion
This attempts a general description of the process whereby a relying party checks whether they will accept Verifiable Credential or Verifiable Presentation. An additional goal is to provide a framework and context for the “verification” steps provided in the vc-http-api repository.
```

File: docs/verification.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+  - [Verification](#verification)
+  - [Validation](#validation)
+  - [Veracity](#veracity)
+- [Interfaces between Validation and Verification](#interfaces-between-validation-and-verification)
+- [Sources](#sources)
+
+
+# Definitions
+
+Definitions of the first two terms come from the [VC-DATA-MODEL](https://www.w3.org/TR/vc-data-model). The third is a replacement/repurposing of what the VC-DATA-MODEL calls “Validity”. (Renaming due to insufficient distinction between the words “validation” and “validity”.)
```
Comment (msporny) [inline docs/verification.md:22]: Putting the reasoning here is fine for the time being, but ultimately, we should fix the VC spec.

File: docs/verification.md [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+2. Make sure the `domain` and `challenge` values are ones the verifier was expecting
+
+
+## Validation
+
+These are categories of validation checks, but do not prescribe specific ways of checking. The verifier may determine how these checks are performed by providing callbacks to the “Verification” layer. This interface is described in "Interfaces between Validation and Verification".
+
+*   Timeliness and Status (callback to verification)
+    *   issuanceDate isn’t in the future, not past expiration date
+    *   Status is “current”, e.g. not revoked
```
Comment (TallTed) [inline docs/verification.md:81]:
```suggestion
    *   Status is “current”, e.g., not revoked, not expired, etc.
```

File: docs/verification.md [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+    *   A more complete list of considerations should be provided in the context of the DID spec (or related docs). Here are some examples:
+        *   Check to make sure the signing keys are not marked revoked in the latest version of the DID document (or have not been removed).
+            *   If the keys _are_ currently revoked, try and determine whether they were valid at the time the VC was issued, and whether to accept it. (This is very much usecase/threat-model specific).
+            *   Different DID methods and Key types have different ways of marking key expiration/revocation.
+        *   Check to make sure the issuer DID has not been revoked.
+
+
+## Veracity
+
+Perform any other app-specific or domain-specific validation around fitness for purpose.
```
Comment (msporny) [inline docs/verification.md:105]:
```suggestion
Perform any other application-specific or domain-specific validation around fitness for purpose. 
```

File: docs/verification.md [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+    *   Status is “current”, e.g. not revoked
+*   Credential Holder/Subject is expected party
+    *   Authenticate 
+        *   Including DID auth, traditional auth method
+    *   Other checks
+        *   Public key material (TODO: check this doesn't conflict with another category) 
+        *   Subject attributes listed in the credential 
+*   Issuer is expected party
+    *   The value associated with the `issuer` property identifies an issuer that is known to and trusted by the verifier.
+        *   E.g. Check the issuer DID against an allow-list
```
Comment (TallTed) [inline docs/verification.md:90]:
```suggestion
        *   E.g., Check the issuer DID against an allow-list
```

File: docs/verification.md [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+        *   Check to make sure the issuer DID has not been revoked.
+
+
+## Veracity
+
+Perform any other app-specific or domain-specific validation around fitness for purpose. 
+
+Veracity checks do not fall in the scope of vc-http-api, but we call it out as a general bucket for additional checks, which serve one of these purposes
+
+1. Restrict the credentials accepted, to establish credibility, reputation, or “premium brand presence”. Examples include:
```
Comment (msporny) [inline docs/verification.md:109]: what does "premium brand presence" mean?

File: docs/verification.md [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+    1. Accept credentials with specific accreditations
+    2. Checking nested signatures/proofs
+2. Relax the credential accepted: i.e. override a previous “failed” status, to accept a broader range of credentials. Examples may include:
+    3. Accept issuanceDate in the future (post-dated checks)
+    4. Accept another institution’s expired credentials (e.g. if the issuing institution went out of business)
```
Comment (msporny) [inline docs/verification.md:114]:
```suggestion
    1. Accept issuanceDate in the future (post-dated checks)
    2. Accept another institution’s expired credentials (e.g. if the issuing institution went out of business) 
```

**Review by msporny — APPROVED** [review]
This feels like it is going to turn into a spec... or belong in vc-imp-guide? https://w3c.github.io/vc-imp-guide/

Minor optional nits and suggestions, otherwise LGTM. Thank you @kimdhamilton !

File: docs/verification.md [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+            *   If the keys _are_ currently revoked, try and determine whether they were valid at the time the VC was issued, and whether to accept it. (This is very much usecase/threat-model specific).
+            *   Different DID methods and Key types have different ways of marking key expiration/revocation.
+        *   Check to make sure the issuer DID has not been revoked.
+
+
+## Veracity
+
+Perform any other app-specific or domain-specific validation around fitness for purpose. 
+
+Veracity checks do not fall in the scope of vc-http-api, but we call it out as a general bucket for additional checks, which serve one of these purposes
```
Comment (TallTed) [inline docs/verification.md:107]:
```suggestion
Veracity checks do not fall within the scope of vc-http-api, but we call it out as a general bucket for additional checks, which serve one of these purposes
```

File: docs/verification.md [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+
+1. Restrict the credentials accepted, to establish credibility, reputation, or “premium brand presence”. Examples include:
+    1. Accept credentials with specific accreditations
+    2. Checking nested signatures/proofs
+2. Relax the credential accepted: i.e. override a previous “failed” status, to accept a broader range of credentials. Examples may include:
```
Comment (TallTed) [inline docs/verification.md:112]:
```suggestion
2. Relax the credential accepted: e.g., override a previous “failed” status, to accept a broader range of credentials. Examples may include:
```

File: docs/verification.md (hunk 8 above)
Comment (TallTed) [inline docs/verification.md:114]:
```suggestion
    4. Accept another institution’s expired credentials (e.g., if the issuing institution went out of business) 
```

File: docs/verification.md [hunk 11]
```diff
@@ -0,0 +1,146 @@
...
+    3. Accept issuanceDate in the future (post-dated checks)
+    4. Accept another institution’s expired credentials (e.g. if the issuing institution went out of business) 
+
+# Interfaces between Validation and Verification
+
+There are two relevant interfaces between Validation and Verification:
+
+*   Callbacks to determine credential timeliness and status
+    *   This provides flexibility for verifier to make their own judgment about whether they’ll accept a credential; i.e. some parties may accept an issuanceDate in the future but others won’t
```
Comment (TallTed) [inline docs/verification.md:121]:
```suggestion
    *   This provides flexibility for verifier to make their own judgment about whether they’ll accept a credential; e.g., some parties may accept an issuanceDate that falls in the future, while others won’t
```

**Review by OR13 — APPROVED** [review]

File: docs/verification.md (hunk 3 above)
Comment (kimdhamilton) [reply docs/verification.md:22]: +1

**Comment by kimdhamilton** [comment]
> This feels like it is going to turn into a spec (a good thing)... or belong in vc-imp-guide? https://w3c.github.io/vc-imp-guide/
> 
> Minor optional nits and suggestions, otherwise LGTM. Thank you @kimdhamilton !

Thanks Manu! I'd be happy with it going to either destination long term. Just wanted to capture the process somewhere to start with, so we can keep iterating on it in the meantime.

File: docs/verification.md (hunk 7 above)
Comment (kimdhamilton) [reply docs/verification.md]: I reworded it slightly.

**Pull request merged**